packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -0,0 +1,142 @@
+import { useCallback, useRef } from 'react';
+import styled from '@emotion/styled';
+import { useIntl } from 'react-intl';
+import { useShowNotification } from '@commercetools-frontend/actions-global';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import {
+  DOMAINS,
+  NOTIFICATION_KINDS_PAGE,
+} from '@commercetools-frontend/constants';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import CustomPanel from '../custom-panel/custom-panel';
+import messages from './messages';
+
+type TCustomViewIframeMessage = {
+  source: string;
+  destination: string;
+  eventName: string;
+  eventData: Record<string, unknown>;
+};
+
+/*
+  TODO: These types are temporary until we have the proper
+  ones generated from the Settings schema
+*/
+export type TPermissionGroup = {
+  name: string;
+  oAuthScopes: string[];
+};
+
+export type TCustomView = {
+  id: string;
+  defaultLabel: string;
+  labelAllLocales: Record<string, string>;
+  url: string;
+  type: 'custom-panel' | 'custom-tab';
+  typeConfig?: {
+    size?: 'small' | 'large';
+  };
+  locators: string[];
+};
+
+type TCustomViewLoaderProps = {
+  customView: TCustomView;
+  onClose: () => void;
+};
+
+const isIframeReady = (iFrameElementRef: HTMLIFrameElement) => {
+  try {
+    return iFrameElementRef?.contentWindow?.document.readyState === 'complete';
+  } catch {
+    // Trying to access the contentWindow of a cross-origin iFrame will throw an error.
+    // We are not supposed to even get here because the iFrame must use
+    // a URL from our very same domain (the custom view is proxied through our http-proxy service).
+    return false;
+  }
+};
+
+const CustomPanelIframe = styled.iframe`
+  height: 100%;
+  width: 100%;
+  border: none;
+`;
+
+function CustomViewLoader(props: TCustomViewLoaderProps) {
+  const iFrameElementRef = useRef<HTMLIFrameElement>(null);
+  const appContext = useApplicationContext();
+  const intercomChannel = useRef(new MessageChannel());
+  const showNotification = useShowNotification();
+  const intl = useIntl();
+
+  const messageFromIFrameHandler = useCallback((event: MessageEvent) => {
+    if (event.data.origin === window.location.origin) {
+      console.log('message received from iframe port: ', event);
+    }
+  }, []);
+
+  // onLoad handler is called from the iFrame even where the URL is not valid
+  // (blocked by CORS, 404, etc.) so we need to make sure the iFrame is ready
+  const onLoadSuccessHandler = useCallback(() => {
+    // Show error and block if the iFrame is not ready
+    // (error loading it)
+    if (!isIframeReady(iFrameElementRef.current!)) {
+      showNotification({
+        domain: DOMAINS.PAGE,
+        kind: NOTIFICATION_KINDS_PAGE.error,
+        text: intl.formatMessage(messages.loadError),
+      });
+      return;
+    }
+
+    // Listen for messages from the iFrame
+    intercomChannel.current.port1.onmessage = messageFromIFrameHandler;
```

Comment:
We're not expecting the iFrame to send any message in the MVP.
I left this part here mostly for reference purposes but maybe we should remove it.